### PR TITLE
Utilities for calculating project stats

### PR DESCRIPTION
**Pull request by noahsark769**
Added in a class called `KilowattStatsAggregator` to help calculate project stats that we'll need to use for #108 and #109.

```python
>>> Project.factories.base.create(impact_power=1.0).statistics.as_dict()
{'automobile_miles_per_month': 263.46399880275374, 'pounds_carbon_saved_per_month': 673.87149, 'acres_of_trees_saved_per_year': 0.855200975609756, 'dollars_saved_per_month': 55.516784}
```

@sameerav or @vivekraghuram or @erictshen can one of you make sure this looks okay?

### CODE REVIEW

File: revolv/project/stats.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    NUM_HOURS_IN_YEAR = 8765.81
+    POUNDS_CARBON_PER_KWH = 1.845
+    KWH_PER_ACRE_OF_TREES = 5125
+    KWH_PER_GALLON_GAS = 33.41
+    # the average mpg number may need to periodically be updated, as it is
+    # prone to fast increase!
+    AVERAGE_AMERICAN_MPG = 24.1
+    # percentage of day the solar panels produce power
+    # roughly 11.5 hours a day in CA: http://aa.usno.navy.mil/cgi-bin/aa_durtablew.pl?form=1&year=2015&task=-1&state=CA&place=San+Francisco
+    SOLAR_PANEL_USE_FACTOR_PER_DAY = 11.5 / 24
```
Comment (erictshen) [inline revolv/project/stats.py]: Are all of these from Andreas?

File: revolv/project/stats.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    to engage donors, it makes sense for RE-volv to display the impact of a solar
+    project in terms of not only power output but dollars saved per month, equivalent
+    acres of trees that could be planted, etc.
+
+    This is a utility class that takes in a kilowatt number (or, can accept instantiation
+    from a Project or a Project queryset) and makes different statistics about the
+    kilowatt power output available through calculations. Another advantage is that if
+    we need to change the calculations, the price of electricity, or anything else
+    related to the project stats, we can do so here and only here.
+    """
```
Comment (erictshen) [inline revolv/project/stats.py]: Is there a constructor that takes in number of kilowatt hours?

File: revolv/project/stats.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    def from_project_queryset(cls, queryset):
+        """Instantiate an aggregator from a queryset of Projects."""
+        total_kilowatts = queryset.aggregate(Sum("impact_power"))["impact_power__sum"]
+
+        if total_kilowatts is None:
+            raise KilowattStatsAggregatorException("Could not determine total kilowatt output of Project queryset.")
+
+        return cls(queryset.aggregate(Sum("impact_power"))["impact_power__sum"])
+
+    def __init__(self, kilowatts):
```
Comment (erictshen) [inline revolv/project/stats.py]: Maybe add a quick docstring, but it's very basic so not really a need to.

File: revolv/project/stats.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+"""
+from django.db.models import Sum
+
+
+class KilowattStatsAggregatorException(Exception):
+    pass
+
+
+class KilowattStatsAggregator(object):
+    """
```
Comment (erictshen) [inline revolv/project/stats.py]: Are these kilowatt hours or just kilowatts? And maybe link to a page that defines what specifically they are in regards to solar. Even after a year, still not fully sure of the distinction =P

**Comment by erictshen** [comment]
Code LGTM, some minor nitpicks about adding more comments, but up to you about whether to add them.

File: revolv/project/stats.py (hunk 1 above)
Comment (noahsark769) [inline revolv/project/stats.py]: These are from online sources (see links in docstrings). I was thinking that I would confirm them with Andreas when we meet with him next, but wanted to get this PR up ASAP

File: revolv/project/stats.py (hunk 4 above)
Comment (noahsark769) [inline revolv/project/stats.py]: They're kilowatts, but I agree that we should make sure kilowatt/kWh in docstrings. Will do that!

File: revolv/project/stats.py (hunk 3 above)
Comment (noahsark769) [inline revolv/project/stats.py]: Good idea!

**Comment by noahsark769** [comment]
Yup the comments are important I think, esp dealing with kW/kWh. Will add those then merge this

**Comment by noahsark769** [comment]
Alright, updated docs a lot. Will merge when codeship passes!

**Pull request merged**